src/main/java/org/sunbird/ratings/service/RatingServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+
+    @Autowired
+    Producer producer;
+
+    //TO DO
+    @Override
+    public SBApiResponse getRatings(String activity_Id, String activity_Type, String userId) {
+        return null;
+    }
+
+    //TO DO
+    @Override
+    public SBApiResponse getUsers(String activity_Id, String activity_Type, String userId) {
+        return null;
+    }
+
+    //TO DO
+    @Override
+    public SBApiResponse upsertRating(String activity_Id, String activity_Type, String userId) {
+        return null;
+    }
+
+    //TO DO
+    @Override
+    public SBApiResponse getRatingSummary(String activity_Id, String activity_Type) {
+        return null;
+    }
+
+    @Override
+    public SBApiResponse upsert(RequestRating requestRating) {
+        UUID timeBasedUuid = UUIDs.timeBased();
+
+        SBApiResponse response = new SBApiResponse(Constants.API_RATINGS_ADD);
+        RatingMessage msg = new RatingMessage();
+
+        try {
+            validateRatingsInfo(requestRating);
+
+            Map<String, Object> request = new HashMap<>();
+            request.put(Constants.ACTIVITY_ID, requestRating.getActivity_Id());
+            request.put(Constants.ACTIVITY_TYPE, requestRating.getActivity_type());
+            request.put(Constants.RATINGS_USER_ID, requestRating.getUserId());
+
+
+            List<Map<String, Object>> existingDataList = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
+                    Constants.TABLE_RATINGS, request, null);
+
+            if (!existingDataList.isEmpty()) {
+
+                Map<String, Object> updateRequest = new HashMap<>();
+                updateRequest.put(Constants.RATING, requestRating.getRating());
+                updateRequest.put(Constants.REVIEW, requestRating.getReview());
+                updateRequest.put(Constants.UPDATED_ON, timeBasedUuid);
+
+                Map<String, Object> prevInfo = existingDataList.get(0);
+                cassandraOperation.updateRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_RATINGS, updateRequest,
+                        request);
+
+                msg.setAction("ratingUpdate");
+                msg.setActivity_id(requestRating.getActivity_Id());
+                msg.setActivity_Type(requestRating.getActivity_type());
+                msg.setUser_id(requestRating.getUserId());
+                msg.setCreated_Date(String.valueOf((prevInfo.get("createdon"))));
+
+                UpdatedValues prevValues = processEventMessage(String.valueOf(prevInfo.get("createdon")),
+                        (Float) prevInfo.get("rating"), (String) prevInfo.get("review"));
+
+                UpdatedValues updatedValues = processEventMessage(String.valueOf(updateRequest.get(Constants.UPDATED_ON)),
+                        requestRating.getRating(), requestRating.getReview());
+
+                msg.setPrevValues(prevValues);
+                msg.setUpdatedValues(updatedValues);
+
+                producer.push(Constants.RATINGS_UPDATE_EVENT, msg);
```

Comment:
The topic name should be read from env variables and not from constants. This is required because the topic names would be namespaced with env names like dev.topicname/stage.topicname etc

Reply:
will change it then